samples/hackathon/hack_base/hack_base.cpp

```diff
@@ -224,48 +222,62 @@ void hack_base::update_rotation(float delta_time)
 			auto fy = static_cast<float>(y);
 			for (uint32_t z = 0; z < dim; z++)
 			{
-				auto fz = static_cast<float>(z);
+				auto fz    = static_cast<float>(z);
 				auto index = x * dim * dim + y * dim + z;
 
 				// Update rotations
 				rotations[index] += animation_timer * rotation_speeds[index];
 
 				// Update matrices
 				glm::vec3 pos(-((fdim * offset.x) / 2.0f) + offset.x / 2.0f + fx * offset.x,
-					-((fdim * offset.y) / 2.0f) + offset.y / 2.0f + fy * offset.y,
-					-((fdim * offset.z) / 2.0f) + offset.z / 2.0f + fz * offset.z);
+				              -((fdim * offset.y) / 2.0f) + offset.y / 2.0f + fy * offset.y,
+				              -((fdim * offset.z) / 2.0f) + offset.z / 2.0f + fz * offset.z);
 
 				glm::mat4 *model_mat = get_aligned_cube(index);
-				*model_mat = glm::translate(glm::mat4(1.0f), pos);
-				*model_mat = glm::rotate(*model_mat, rotations[index].x, glm::vec3(1.0f, 1.0f, 0.0f));
-				*model_mat = glm::rotate(*model_mat, rotations[index].y, glm::vec3(0.0f, 1.0f, 0.0f));
-				*model_mat = glm::rotate(*model_mat, rotations[index].z, glm::vec3(0.0f, 0.0f, 1.0f));
+				*model_mat           = glm::translate(glm::mat4(1.0f), pos);
+				*model_mat           = glm::rotate(*model_mat, rotations[index].x, glm::vec3(1.0f, 1.0f, 0.0f));
+				*model_mat           = glm::rotate(*model_mat, rotations[index].y, glm::vec3(0.0f, 1.0f, 0.0f));
+				*model_mat           = glm::rotate(*model_mat, rotations[index].z, glm::vec3(0.0f, 0.0f, 1.0f));
 			}
 		}
 	}
 }
 
 void hack_base::draw()
 {
-	ApiVulkanSample::prepare_frame();
+	ScopedTiming _(stopwatch, (int) MeasurementPoints::FullDrawCall);
 
-	// Command buffer to be submitted to the queue
-	submit_info.commandBufferCount = 1;
-	submit_info.pCommandBuffers = &draw_cmd_buffers[current_buffer];
+	{
+		ScopedTiming _(stopwatch, (int) MeasurementPoints::PrepareFrame);
```

Comment:
We should capture `hack_prepare();` instead inside `prepare` as the rest is just base class overhead that doesn't change.